src/entry.rs

```diff
@@ -489,6 +495,13 @@ impl<'a> EntryFields<'a> {
             return Ok(());
         };
 
+        // Old BSD-tar compatibility.
+        //
+        // Names that have a trailing slash should be treated as a directory.
+        if truncate(&self.header.as_old().name).ends_with(b"/") {
```

Comment:
Could this perhaps use something like `name_bytes()` so the `truncate` and `as_old` and such is already done internally?

Reply:
Done!